spdb/spatialdb/annocube.py

```diff
@@ -92,6 +92,37 @@ def overwrite(self, input_data, time_sample_range=None):
             self.data[time_sample_range[0], :, :, :] = ndlib.overwriteDense64_ctype(
                 self.data[time_sample_range[0], :, :, :], input_data[time_sample_range[0], :, :, :])
 
+    def overwrite_to_black(self, input_data, time_sample_range=None):
+        """ Overwrite data with zero values in the input_data
+
+        Function is accelerated via ctypes lib.
+
+        If time_sample_range is provided, data will be inserted at the appropriate time sample
+
+        Args:
+            input_data (numpy.ndarray): Input data matrix to overwrite the current Cube data
+            time_sample_range list(int): The min and max time samples that input_data represents in python convention
+            (start inclusive, stop exclusive)
+
+        Returns:
+            None
+
+        """
+        if self.data.dtype != input_data.dtype:
+            raise SpdbError("Conflicting data types for overwrite.",
+                            ErrorCodes.DATATYPE_MISMATCH)
+
+        if not time_sample_range:
+            # If no time sample range provided use default of 0
+            time_sample_range = [0, 1]
+
+        if input_data.ndim == 4:
+            for t in range(*time_sample_range):
+                self.data[t, :, :, :] = np.where(input_data[t - time_sample_range[0], :, :, :]==1, self.data[t, :, :, :]==0, self.data[t, :, :, :])
+        else:
+            # Input data doesn't have any time indices
+            self.data[time_sample_range[0], :, :, :] = np.where(input_data[time_sample_range[0], :, :, :]==1, self.data[t, :, :, :]==0, self.data[t, :, :, :])
```

Comment:
I would look at other places in the code to see how long lines are, but it would make it a little easier to read if you break this into a multi-line statement. (and the statement above, in the loop).